File: dataset/codenet/Python/Code/s793709854.py
N,T = map(int, input().split())
L = [[int(l) for l in input().split()] for _ in range(N)]

dp = [[0]*T for _ in range(N+1)]

for i in range(N):
    for j in range(T):
        if j < L[i][0]:
            dp[i+1][j] = dp[i][j]
        else:
            dp[i+1][j] = max(dp[i][j], dp[i][j-L[i][0]]+L[i][1])
            
used = []
B = dp[N][T-1]
j = T-1
for i in range(N-1, -1, -1):
    if B == dp[i][j]:
        continue
    B -= L[i][1]
    used.append(i)
    j -= L[i][0]
        
ans = dp[N][T-1]
c = 0
for i in range(N):
    if i in used:
        continue
    c = max(c, L[i][1])
    
print(ans+c)

File: output/qwen_2.5_72b_instruct/codenet/algo_based_translation/python/java/s793709854.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class s793709854 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int T = scanner.nextInt();
        List<List<Integer>> L = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            List<Integer> item = new ArrayList<>();
            item.add(scanner.nextInt());
            item.add(scanner.nextInt());
            L.add(item);
        }
        int[][] dp = new int[N + 1][T];
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < T; j++) {
                if (j < L.get(i).get(0)) {
                    dp[i + 1][j] = dp[i][j];
                } else {
                    dp[i + 1][j] = Math.max(dp[i][j], dp[i][j - L.get(i).get(0)] + L.get(i).get(1));
                }
            }
        }
        List<Integer> used = new ArrayList<>();
        int B = dp[N][T - 1];
        int j = T - 1;
        for (int i = N - 1; i >= 0; i--) {
            if (B == dp[i][j]) {
                continue;
            }
            B -= L.get(i).get(1);
            used.add(i);
            j -= L.get(i).get(0);
        }
        int ans = dp[N][T - 1];
        int c = 0;
        for (int i = 0; i < N; i++) {
            if (used.contains(i)) {
                continue;
            }
            c = Math.max(c, L.get(i).get(1));
        }
        System.out.println(ans + c);
    }
}
